src/app/features/theme-drawer/theme-list-item.tsx

```diff
@@ -0,0 +1,33 @@
+import { useCallback } from 'react';
+import { BoxProps } from '@stacks/ui';
+
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { getThemeLabel, UserSelectedTheme, useThemeSwitcher } from '@app/common/theme-provider';
+import { ThemeListItemLayout } from './theme-list-item-layout';
+
+interface ThemeListItemProps extends BoxProps {
+  theme: UserSelectedTheme;
+}
```

Comment:
Keeping the logic in each `<ThemeListItem />` works, but we could make it more generic with [named event handlers](https://beta.reactjs.org/learn/responding-to-events#naming-event-handler-props).

The benefit of this is that `ThemeListItem` remains dumb, only working with input/outputs. This also makes it easier to handle a case where say, one requires slightly different behaviour. 

Although small, we could consider the fact that it's a click event that changes the switcher, an implementation detail we abstract when moving the logic out.

```suggestion
interface ThemeListItemProps extends BoxProps {
  theme: UserSelectedTheme;
  onThemeSelected(theme: UserSelectedTheme): void;
}
```

Reply:
Alright, will make it a dumb component.